Sources/ApolloSQLite/SQLiteNormalizedCache.swift

```diff
@@ -140,11 +140,12 @@ extension SQLiteNormalizedCache: NormalizedCache {
     let result: Swift.Result<[Record?], Error>
     do {
       let records = try self.selectRecords(forKeys: keys)
+      let recordsIndexMap = records.indices.reversed().reduce(into: [:]) { resultMap, index in
```

Comment:
What's the reason for the reverse iteration here? We're not mutating `records`, so I don't think iteration order should matter.

Reply:
The old snippet was for some reason picking the *first* record that matches the key. I wanted to keep the behavior as is and `reversed` as per [Apple documentation](https://developer.apple.com/documentation/swift/array/1690025-reversed)  _"reverse a collection without allocating new space for its elements by calling this reversed() method"_. Hence it is a pretty lightweight operation.

Reply:
Ah, thanks for being careful, but I don't think that is needed here. There can never be more than one record with the same key returned, that is even [being enforced at the database level](https://github.com/apollographql/apollo-ios/blob/293dd3b3287413a057e83050a31d181131b74c7f/Sources/ApolloSQLite/SQLiteNormalizedCache.swift#L70).

Reply:
Ah nice, that makes sense. Let me remove it then.

Reply:
Took care